cmd/clusterctl/test/e2e/go.mod

```diff
@@ -3,21 +3,20 @@ module sigs.k8s.io/cluster-api/cmd/clusterctl/test/e2e
 go 1.13
```

Comment:
Can we remove this submodule altogether?

Reply:
definitely

Reply:
It looks like if we do that we'll be adding a dependency on sigs.k8s.io/cluster-api/test/infrastructure/docker to cluster-api proper.

Reply:
What about the docker one? Doesn't look massively different atm.

Reply:
we're trying to keep capd separate until we have a better story for DX https://github.com/kubernetes-sigs/cluster-api/issues/1565#issuecomment-594974850. 

If this pulls CAPD in cluster-api, then it should probably stay in a sub-module